pkg/sink/elasticsearch.go

```diff
@@ -174,38 +172,29 @@ func (e *Elasticsearch) flushIndex(ctx context.Context, indexCfg config.ELSIndex
 	return nil
 }
 
-// SendEvent sends event notification to Elasticsearch
-func (e *Elasticsearch) SendEvent(ctx context.Context, event event.Event, eventSources []string) (err error) {
-	e.log.Debugf(">> Sending to Elasticsearch: %+v", event)
+// SendMessageToAll is no-op.
+func (e *Elasticsearch) SendMessageToAll(_ context.Context, _ interactive.CoreMessage) error {
+	return nil
+}
+
+// SendMessage sends event notification to Elasticsearch.
+func (e *Elasticsearch) SendMessage(ctx context.Context, _ interactive.CoreMessage, _ []string, rawData any) error {
+	e.log.Debugf(">> Sending to Elasticsearch: %+v", rawData)
 
 	errs := multierror.New()
 	for _, indexCfg := range e.indices {
-		if !sliceutil.Intersect(indexCfg.Bindings.Sources, eventSources) {
-			continue
-		}
-
-		err := e.flushIndex(ctx, indexCfg, event)
+		err := e.flushIndex(ctx, indexCfg, rawData)
```

Comment:
That's might be problematic as I know we needed to ignore some properties from the `event.Event` as ELS didn't accept some of the nested struct data. Just saying, but it should work OK as long as all events have that in mind 👍

Reply:
I tested this for kubernetes plugin and the data was shipped to elasticsearch, but I think we need to state this in the docs maybe, "we don't have control on data structure, so if you have es integration, you might get error due to your complex nested data structure coming from custom plugin".